### ISSUE TITLE
Bug: Style/AccessModifierDeclarations fails with certain syntax

### ISSUE THREAD

**Issue by johnnyshields** [rubocop/rubocop#11102]
Given this code:

```ruby
module MyModule
  extend self

  def my_method
  end
  
  singleton_methods.each { |method| module_function(method) }
end
```

Running `rubocop` produces this error:

```ruby
An error occurred while Style/AccessModifierDeclarations cop was inspecting C:/workspace/my_file.rb.
Errors are usually caused by RuboCop bugs.
Please, report your problems to RuboCop's issue tracker.
https://github.com/rubocop/rubocop/issues

Mention the following information in the issue report:
1.37.0 (using Parser 3.1.2.1, rubocop-ast 1.23.0, running on ruby 3.1.1) [x64-mingw-ucrt]
```
